websites/permissions.py

```diff
@@ -0,0 +1,245 @@
+        current_perms = set(get_perms(group_or_user, website))
+    elif isinstance(group_or_user, Group):
+        current_perms = {perm.codename for perm in group_or_user.permissions.all()}
+    elif isinstance(group_or_user, User):
+        current_perms = {
+            perm.codename for perm in Permission.objects.filter(user=group_or_user)
+        }
+    else:
+        raise TypeError("Permissions must be assigned to a user or group")
+    if not {perm.replace("websites.", "") for perm in perms}.difference(current_perms):
+        return False
+    for perm in perms:
+        try:
+            if website:
+                assign_perm(perm, group_or_user, website)
+            else:
+                assign_perm(perm, group_or_user)
+        except Permission.DoesNotExist as err:
+            raise Permission.DoesNotExist(f"Permission '{perm}' not found") from err
+    return True
+
+
+@transaction.atomic
+def create_website_groups(website):
+    """
+    Create groups and assign permissions for a website
+
+    Args:
+        website (Website): The website that the permissions are for
+
+    Returns:
+        list (int, int, bool): # groups created, # groups updated, owner updated
+    """
+    groups_created = 0
+    groups_updated = 0
+    owner_updated = False
+
+    admin_group, admin_created = Group.objects.get_or_create(
+        name=f"{constants.ADMIN_GROUP}{website.uuid.hex}"
+    )
+    if admin_created:
+        groups_created += 1
+    editor_group, editor_created = Group.objects.get_or_create(
+        name=f"{constants.EDITOR_GROUP}{website.uuid.hex}"
+    )
+    if editor_created:
+        groups_created += 1
+
+    if (
+        assign_website_permissions(
+            admin_group, constants.PERMISSIONS_ADMIN, website=website
+        )
+        and not admin_created
+    ):
+        groups_updated += 1
+
+    if (
+        assign_website_permissions(
+            editor_group, constants.PERMISSIONS_EDITOR, website=website
+        )
+        and not editor_created
+    ):
+        groups_updated += 1
+
+    if website.owner:
+        owner_updated = assign_website_permissions(
+            website.owner, constants.PERMISSIONS_ADMIN, website=website
+        )
+
+    return groups_created, groups_updated, owner_updated
+
+
+@transaction.atomic
+def create_global_groups():
+    """
+    Create the global groups for website permissions
+
+    Args:
+        website (Website): The website that the permissions are for
+
+    Returns:
+        list (int, int): # groups created, # groups updated
+    """
+    groups_created = 0
+    groups_updated = 0
+
+    admin_group, admin_created = Group.objects.get_or_create(
+        name=constants.GLOBAL_ADMIN
+    )
+    if admin_created:
+        groups_created += 1
+    author_group, author_created = Group.objects.get_or_create(
+        name=constants.GLOBAL_AUTHOR
+    )
+    if author_created:
+        groups_created += 1
+
+    global_perms = [constants.PERMISSION_ADD]
```

Comment:
What's the significance of this? Should it be represented in constants.py? We have `PERMISSIONS_EDITOR` and `PERMISSIONS_ADMIN`. Should we also have `PERMISSIONS_AUTHOR` or something like that?

Reply:
This is the only permission a global author has - creating new websites (but then will have all the admin permissions of any site created, as the owner of that site).  I didn't create a constant for it because I felt it would be a little redundant, since it's just another constant in a list (`constants.PERMISSION_ADD`)